### ISSUE TITLE
Small freezes in World of Warcraft sometimes

### ISSUE THREAD

**Issue by Thaodan** [Frogging-Family/wine-tkg-git#254]
* Issue
Sometimes I get small freezes for like 1-3s in WoW, the rest of the system is unaffected. The game still responds but the UI/graphics and Audio is frozen.

* Buildinfo
Stock wine-tkg-git (6.0rc4.r2.g8b1e0eec) build in a clean chroot.

* Additional Information
I read here [1] that esync helps when wineserver pegs a core. So maybe my core count is relevant as I have an 32 Core threadripper.

-----
[1] https://github.com/PhoenicisOrg/phoenicis-winebuild/issues/57

**Comment by Thaodan** [comment]
I noticed every time this happens this logged:
```
02cc:err:sync:RtlpWaitForCriticalSection section <same address> "?" wait timed out in thread 02cc, blocked by 0000, retrying (60 sec)
```

**Comment by Thaodan** [comment]
I noticed that the thread is not always the same thread that blocks.

Downgrading to 6.0rc3.r1.g7ad10427-312 works the issue around.

**Comment by das-bub** [comment]
I'm seeing the same in Deus Ex: Human Revolution with 6.0rc4.r3.g0e67af3d-312 on my R5 3600.

Likely caused by [this upstream commit](https://github.com/wine-staging/wine-staging/commit/b8ca0eae9f47491ba257c422a2bc03fc37d13c22).

**Comment by das-bub** [comment]
I've tested it just now, downgrading to 6.0rc4.r0.g350eb136-312 fixes the issue for me.

**Comment by Thaodan** [comment]
Did you test it by building without that patch on the recent version?

**Comment by Thaodan** [comment]
Reported the issue upstream here:
https://bugs.winehq.org/show_bug.cgi?id=50448

**Comment by das-bub** [comment]
The issue is fixed both when compiling 6.0rc4 before the first patch, so commit 350eb13, and when compiling 6.0rc5 with the patches reverted.

**Comment by Thaodan** [comment]
> The issue is fixed both when compiling 6.0rc4 before the first patch, so
> commit 350eb13, and when compiling 6.0rc5 with the patches reverted.
> 

Please look into the upstream bug and report there.

**Comment by das-bub** [comment]
I already did, but can't provide any input until at least Friday, sorry.

**Comment by ibrokemypie** [comment]
The easiest way to apply the [patch](https://bugs.winehq.org/attachment.cgi?id=69082&action=edit) from the discussion upstream is to download it to the "wine-tkg-userpatches" directory with the extension ".mypatch"

eg
``wget "https://bugs.winehq.org/attachment.cgi?id=69082&action=diff&context=patch&collapsed=&headers=1&format=raw" -O cs-wake-all.mypatch`` from within the directory, and choose yes in the prompt when building wine


The patch does solve the freezes for me as of right now.

**Comment by Tk-Glitch** [comment]
I have decided to disable the ntdll-NtAlertThreadByThreadId staging patch for now (https://github.com/Frogging-Family/wine-tkg-git/commit/5451ab5576f50011daa879e3196e0ab7e246f540). This should fix the issue, hopefully.

**Comment by Thaodan** [comment]
There is already a fix for that, it 
just has to applied to upstream.

**Comment by Tk-Glitch** [comment]
The fix doesn't resolve the issue (and it's already applied on -tkg via the hotfixer), it just reduces it, reportedly. Disabling the patchset altogether fixes it along with the other performance regressions it introduces.

**Comment by eNTi** [comment]
It seems fixed for me... I didn't notice any more freezes with 6.0r0...

**Comment by Tk-Glitch** [comment]
It changes per app/game. WoW might not be very affected. Other games will see a firm performance drop with that patchset (which is very well known by Zeb since he asked me to test it and hesitated for a few weeks before merging in staging). Titanfall 2 sees up to a 5% drop, and Star Citizen is more like 15%. Considering the patch is only making the code cleaner and doesn't improve stability or compatibility, there is no reason to keep it enabled when we are chasing the best performance for gaming. If the gap can be narrowed or even closed, or if it becomes a dependency for another patchset that could be desirable, it'll be re-enabled.

**Comment by eNTi** [comment]
Of course. I wasn't really arguing for the code to be kept. I just wanted to give my experience with a particular game. It was quite bad before the patch... having 2-5s complete game lockups at random times.

**Comment by Tk-Glitch** [comment]
Oh yeah sure :smile: I guess I can safely close.

**Comment by Thaodan** [comment]
> The fix doesn't resolve the issue (and it's already applied on -tkg via the
> hotfixer), it just reduces it, reportedly. Disabling the patchset
> altogether fixes it along with the other performance regressions it
> introduces. 

Upstream says it is fixed by f8ce6cbb21e09c80320514770c102a1a8e9cfde6.

**Comment by Tk-Glitch** [comment]
I am aware, but it only fixes the huge pauses and not the underlying perf regression.
